packages/rest/src/router/routing-table.ts

```diff
@@ -127,6 +128,8 @@ export class RoutingTable {
       );
     }
     this._routes.push(route);
+    // Sort routes
+    this._routes.sort(compareRoute);
```

Comment:
What benefit do we get by sorting the routes?

Reply:
Think about this example:
```ts
/**
 * A simple controller to bounce back http requests
 */
export class PingController {
  constructor(@inject(RestBindings.Http.REQUEST) private req: Request) {}

  @get('/ping/:me')
  pingMe(@param.path.string('me') me: string): object {
    return {
      greeting: `Hello from ${me}`,
    };
  }

  // Map to `GET /ping`
  @get('/ping')
  ping(): object {
    return {
      greeting: 'Hello from LoopBack',
      date: new Date(),
      url: this.req.url,
      headers: Object.assign({}, this.req.headers),
    };
  }

  @get('/ping/xyz')
  pingXyz(): object {
    return {
      greeting: 'Hello from XYZ',
    };
  }
}
```

I would expect `GET /ping/xyz` is routed to `pingXyz()` instead of `pingMe()`. Without this PR, it matches `GET /ping/:me`.

Reply:
I'll add an integration test.